src/main/java/gov/cabinetofice/gapuserservice/service/DepartmentService.java

```diff
@@ -20,4 +22,16 @@ public List<DepartmentDto> getAllDepartments() {
                 .map(departmentMapper::departmentToDepartmentDto)
                 .collect(Collectors.toList());
     }
+
+    public Department getDepartmentById(int id) {
+        return departmentRepository.findById(id).orElseThrow(() -> new DepartmentNotFoundException("Department not found"));
```

Comment:
Incase this needs reusing - often better to return the optional, and handle the error in the controller so you can be a bit more specific about where this issue was thrown

Reply:
Done that